### PR TITLE
2152 upgrade dotnet

### PR DESCRIPTION
**Pull request by sahma19**

### CODE REVIEW

File: domains/certificates/Query.API/API/Dockerfile [hunk 1]
```diff
@@ -22,4 +22,6 @@ WORKDIR /app
 COPY --from=publish /app/publish .
 COPY migrations/API.sql /migrations/API.sql
 
+ENV ASPNETCORE_HTTP_PORTS=80
```
Comment (tnickelsen) [inline domains/certificates/Query.API/API/Dockerfile:25]: Why this now?

File: domains/certificates/Query.API/API/GranularCertificateIssuer/WalletException.cs [hunk 2]
```diff
@@ -1,16 +1,10 @@
 using System;
-using System.Runtime.Serialization;
 
 namespace API.GranularCertificateIssuer;
 
-[Serializable]
```
Comment (tnickelsen) [inline domains/certificates/Query.API/API/GranularCertificateIssuer/WalletException.cs:6]: Why has this been removed?

File: domains/measurements/Measurements.API/Tests/HealthControllerTests.cs [hunk 3]
```diff
@@ -3,7 +3,7 @@
 using Microsoft.AspNetCore.Mvc.Testing;
 using Xunit;
 
-namespace Integration.Tests.Controllers;
+namespace Tests;
```
Comment (tnickelsen) [inline domains/measurements/Measurements.API/Tests/HealthControllerTests.cs:6]: Why this change?

File: domains/certificates/Query.API/API/Dockerfile (hunk 1 above)
Comment (sahma19) [reply domains/certificates/Query.API/API/Dockerfile:25]: Morten, from Heimdal, mentioned that this is needed, when upgrading to DOTNET 8.

File: domains/measurements/Measurements.API/Tests/HealthControllerTests.cs (hunk 3 above)
Comment (sahma19) [reply domains/measurements/Measurements.API/Tests/HealthControllerTests.cs:6]: Because the folder is named "Tests" and is reciding in the Measurements.API/Tests/ folder. I believe this project, hasn't been updated in a LONG time.

File: domains/certificates/Query.API/API/GranularCertificateIssuer/WalletException.cs (hunk 2 above)
Comment (sahma19) [reply domains/certificates/Query.API/API/GranularCertificateIssuer/WalletException.cs]: A mistake. Added it back: https://github.com/Energinet-DataHub/energy-origin/pull/521/commits/8b82fef0827618c5f917f6e7e6e082a3885fb9a4

File: domains/certificates/Query.API/API/Dockerfile (hunk 1 above)
Comment (sahma19) [reply domains/certificates/Query.API/API/Dockerfile:25]: It is required when upgrading to dotnet 8

File: domains/certificates/Query.API/API/GranularCertificateIssuer/WalletException.cs [hunk 4]
```diff
@@ -9,8 +8,4 @@ public class WalletException : Exception
     public WalletException(string message) : base(message)
     {
     }
-
-    protected WalletException(SerializationInfo info, StreamingContext context) : base(info, context)
```
Comment (tnickelsen) [inline domains/certificates/Query.API/API/GranularCertificateIssuer/WalletException.cs:13]: Why has this been removed? It was not used or?

Comment (sahma19) [reply domains/certificates/Query.API/API/GranularCertificateIssuer/WalletException.cs:13]: It is a deprecated way, of serialization. as of dotnet 8, using that syntax, it is now obsolete. 
Adding it back produces a warning which, fails pipeline checks:

image

Comment (sahma19) [reply domains/certificates/Query.API/API/GranularCertificateIssuer/WalletException.cs:13]: https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0051

**Review by tnickelsen — APPROVED** [review]
